### PR TITLE
Support browse mode and other in-process scenarios for AMD64 apps when running on ARM64 Windows

### LINKED ISSUE #14397: Browse mode is not available in some electron apps running on Windows 11 ARM
I have observed this problem with the 64 version of 1Password 8, the RIM app by Pneuma solutions and the version of Slack that can be obtained from the Microsoft store running on a virtual machine with Windows 11 ARM. The problem does not occur with the 32 bits version of Slack, the ARM64 version of Visual Studio code or Teams. All the apps work as expected with Narrator. Executing the command focus.devInfo in the Python Console while focus is inside any of the non-working apps shows the IAccessible2 properties related to the object with focus.

### Steps to reproduce:

1. Download and run Slack from the Microsoft Store.
2. While focus is on the app, try turning on browse mode by pressing NVDA +the space bar.

### Actual behavior:

Nothing happens. The following gets added to the lock when the electron app receives focus:

DEBUGWARNING - virtualBuffers.VirtualBuffer.prepare (18:55:41.302) - MainThread (4072):
appModule has no binding handle to injected code, can't prepare virtualBuffer yet.

### Expected behavior:

NVDA can switch between focus and browse mode.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

2022.3.2

#### Windows version:

Edition	Windows 11 Pro

Version	22H2

OS build	22621.819

Experience	Windows Feature Experience Pack 1000.22636.1000.0

#### Name and version of other software in use when reproducing the issue:

N/a

#### Other information about your system:

Windows is running on a Parallels VM

#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #14397

### Summary of the issue:
Although NVDA fully functions on Windows 10 ARM64, including interacting with both ARM64 and x86-emulated apps, on Windows 11 particular in-process NVDA features such as browse mode is not available when interacting with AMD64-emulated apps.
For instance, Running AMD64 builds of Firefox, or 1Password, or Chrome on Windows 11 ARM64, browse mode is not available.
The reason for this is that on ARM64 Windows, NVDA only launches an ARM64 nvdaHelperRemoteLoader process for injecting NVDAHelper, not an AMD64 one. This is because in Windows 10 running AMD64 apps was not supported by the OS.
NVDA should fully support AMD64-emulated apps on windows 11.

### Description of user facing changes
On Windows 11 ARM64, NVDA's browse mode is now supported in AMD64 apps such as Firefox, Google Chrome, and 1Password.

### Description of development approach
* NVDAHelper.py's versionedLib64Path variable has been replaced by versionLibAMD64Path and versionedLibARM64Path variables.
* NVDAHelper.py's RemoteLoader64 class has been renamed to RemoteLoader, and it now takes a loader directory path on construction, so that it is possible to launch NVDAHelperRemoteLoader.exe in either lib64 (AMD64) or libARM64.
* When initializing NVDAHelper, on ARM64, both ARM64 and AMD64 versions of NVDAhelperRemoteLoader are now started, and both are cleaned up on termination.

### Testing strategy:
Ran a try build on Windows 11 ARM64. 
* Opened 1Password (AMD64) and ensured that browse mode was activated for its document.
* Opened Firefox (AMD64) and browsed some web pages, ensuring that browse mode was functioning.
* Opened Microsoft Edge (ARM64) and ensured that browse mode was still functional.
Ran try build on Windows 11 (AMD64):
* Opened Firefox, browsed some web pages and ensured browse mode was astill functional.

### Known issues with pull request:
None known.

### Change log entries:
New features
* On Windows 11 ARM64, browse mode is now available in AMD64 apps such as Firefox, Google Chrome and 1Password. (#14397)
Changes
Bug fixes
For Developers

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by rperez030** [comment]
Thanks @michaelDCurran. It works as expected for me. I wonder if there is any chance this can be included in 2022.4. Too son perhaps? This makes using Windows 11 in arm totally viable for me

**Comment by michaelDCurran** [comment]
Sorry, 2022.4 is already feature complete. Aiming for 2023.1.

File: source/NVDAHelper.py [hunk 1]
```diff
@@ -32,19 +33,20 @@
...
 	versionedLibPath=os.path.join(versionedLibPath,versionInfo.version)
-	versionedLib64Path=os.path.join(versionedLib64Path,versionInfo.version)
+	versionedLibAMD64Path = os.path.join(versionedLibAMD64Path, versionInfo.version)
+	versionedLibARM64Path = os.path.join(versionedLibARM64Path, versionInfo.version)
+
 
 _remoteLib=None
-_remoteLoader64=None
+_remoteLoaderAMD64 = None
+_remoteLoaderARM64 = None
```
Comment (seanbudd) [inline source/NVDAHelper.py:49]: Could these get type hints?

```suggestion
_remoteLoaderAMD64: Optional[RemoteLoader] = None
_remoteLoaderARM64: Optional[RemoteLoader] = None
```

File: source/NVDAHelper.py [hunk 2]
```diff
@@ -481,9 +483,9 @@ def nvdaControllerInternal_openConfigDirectory():
 	return 0
 
 
-class RemoteLoader64(object):
+class RemoteLoader(object):
```
Comment (seanbudd) [inline source/NVDAHelper.py:486]: This appears to be an API breakage?

Should we retain an alias?

```py
class RemoteLoader64(RemoteLoader):
	def __init__(self):
		loaderDir = ...
		super().__init__(loaderDir)
```

Comment (seanbudd) [reply source/NVDAHelper.py:486]: If we want to deprecate this variable, I suggest marking this as private, and also removing the no longer relevant `object` superclass.

```suggestion
class _RemoteLoader:
```

File: source/NVDAHelper.py (hunk 1 above)
Comment (michaelDCurran) [reply source/NVDAHelper.py]: I've added type hints now. But they had to be strings as they appear before the class definition.

File: source/NVDAHelper.py (hunk 2 above)
Comment (michaelDCurran) [reply source/NVDAHelper.py]: RemoteLoader is now private.
Re breaking change: I've mentioned this in changes.t2t under breaking changes. 
I cannot think of any valid (or safe) past external usage of RemoteLoader64. This always should have been (and was somewhat expected that) this was already private. Thus I do not believe adding a backwards compat class is useful.

File: user_docs/en/changes.t2t [hunk 3]
```diff
@@ -86,6 +87,7 @@ Please open a GitHub issue if your Add-on has an issue with updating to the new
     - ``autoTether`` has been removed; ``tetherTo`` can now take the value "auto" instead.
     -
   -
+- The NVDAHelper.RemoteLoader64 class has been removed with no replacement.
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:90]:
```suggestion
- The ``NVDAHelper.RemoteLoader64`` class has been removed with no replacement. (#14449)

```

**Review by seanbudd — APPROVED** [review]

**Comment by seanbudd** [comment]
This API breaking change has been announced here: https://groups.google.com/a/nvaccess.org/g/nvda-api/c/a0v8K2ePY_s/m/JkGsV8G2BQAJ

**Pull request merged**